src/test/java/com/hotels/plunger/ValueAssertionCallStubTest.java

```diff
@@ -0,0 +1,148 @@
+
+  private static final Fields FIELDS = new Fields("field");
+  private static final Fields OUTPUT = new Fields("output");
+
+  private ValueAssertionCallStub<String> stub;
+
+  @Test
+  public void next() {
+    stub = new ValueAssertionCallStub.Builder<String>(FIELDS).addTuple(1).addTuple(2).build();
+    assertThat(stub.getArguments(), is(nullValue()));
+
+    stub.nextAssertionCall();
+    assertThat(stub.getArguments(), is(new TupleEntry(FIELDS, new Tuple(1))));
+
+    stub.nextAssertionCall();
+    assertThat(stub.getArguments(), is(new TupleEntry(FIELDS, new Tuple(2))));
+  }
+
+  @Test
+  public void fields() {
+    stub = new ValueAssertionCallStub.Builder<String>(FIELDS).build();
+    assertThat(stub.getArgumentFields(), is(FIELDS));
+    assertThat(stub.getDeclaredFields(), is(FIELDS));
+  }
+
+  @Test
+  public void argumentsFieldsIgnoreOutputFields() {
+    stub = new ValueAssertionCallStub.Builder<String>(FIELDS).outputFields(OUTPUT).build();
+    assertThat(stub.getArgumentFields(), is(FIELDS));
+    assertThat(stub.getDeclaredFields(), is(OUTPUT));
+  }
+
+  @Test
+  public void context() {
+    stub = new ValueAssertionCallStub.Builder<String>(FIELDS).addTuple(1).addTuple(2).build();
+    assertThat(stub.getContext(), is(nullValue()));
+
+    stub.setContext("VALUE");
+    assertThat(stub.getContext(), is("VALUE"));
+
+    stub.nextAssertionCall();
+    assertThat(stub.getContext(), is("VALUE"));
+  }
+
+  @Test
+  public void collectTupleEntry() {
+    stub = new ValueAssertionCallStub.Builder<String>(FIELDS).build();
+    assertThat(stub.result().asTupleEntryList().isEmpty(), is(true));
+
+    stub.getOutputCollector().add(new TupleEntry(FIELDS, new Tuple(1)));
+    assertThat(stub.result().asTupleEntryList().size(), is(1));
+    assertThat(stub.result().asTupleEntryList().get(0), is(new TupleEntry(FIELDS, new Tuple(1))));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void collectIrregularTupleEntry() {
+    stub = new ValueAssertionCallStub.Builder<String>(FIELDS).build();
+    assertThat(stub.result().asTupleEntryList().isEmpty(), is(true));
+
+    stub.getOutputCollector().add(new TupleEntry(new Fields("X", String.class), new Tuple(1)));
+  }
+
+  @Test
+  public void collectTuple() {
+    stub = new ValueAssertionCallStub.Builder<String>(FIELDS).build();
+    assertThat(stub.result().asTupleEntryList().isEmpty(), is(true));
+
+    stub.getOutputCollector().add(new Tuple(1));
+    assertThat(stub.result().asTupleEntryList().size(), is(1));
+    assertThat(stub.result().asTupleEntryList().get(0), is(new TupleEntry(FIELDS, new Tuple(1))));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void collectIrregularTuple() {
+    stub = new ValueAssertionCallStub.Builder<String>(FIELDS).build();
+    assertThat(stub.result().asTupleEntryList().isEmpty(), is(true));
+
+    stub.getOutputCollector().add(new Tuple(1, 2));
+  }
+
+  // TODO: test withFields()
```

Comment:
When do you plan to do this TODO? (we try keep TODOs to a minimum so I just want to make sure this is planned for soon or could possibly be done now)